<launch>

  #### Nodelet manager ####################################

  <node pkg="nodelet" type="nodelet" name="pelican_manager" args="manager" 
    output="screen" />

  #### Asctec Autopilot ###################################

  <node pkg="nodelet" type="nodelet" name="AutoPilotNodelet" 
    args="load asctec_autopilot/AutoPilotNodelet pelican_manager" output="screen">

<!--     <param name="hum" type="string" value=""/> -->
    <!-- Fica definido uma altura de 5m para os waypoints -->
    <!-- <param name="height" type="int" value="5000"/> -->

    <param name="port" type="string" value="/dev/ttyUSB0"/>
    <!--<param name="port" type="string" value="/dev/pts/10"/>-->
    <param name="speed" type="int" value="57600"/>
    <param name="freq" type="double" value="100.0"/>

    <!-- Set IMU_CALCDATA topic to be true     -->
    <param name="enable_IMU_CALCDATA" type="bool" value="true"/>
    <param name="interval_IMU_CALCDATA" type="int" value="2"/>
    <param name="offset_IMU_CALCDATA" type="int" value="0"/>
    
    <param name="enable_LL_STATUS" type="bool" value="true"/>
    <param name="interval_LL_STATUS" type="int" value="2"/>
    <param name="offset_LL_STATUS" type="int" value="0"/>

    <param name="enable_CONTROL" type="bool" value="false"/>
    <param name="enable_WAYPOINT_COMMANDS" type="bool" value="true"/>
    <param name="enable_GPS_DATA" type="bool" value="true"/>
    <!-- Has to be tested GPS_DATA_ADVACE -->
    <param name="enable_GPS_DATA_ADVANCE" type="bool" value="true"/> 
   
    <param name="enable_CURRENT_WAY" type="bool" value="true"/>
    <param name="interval_CURRENT_WAY" type="int" value="2"/>
    <param name="offset_CURRENT_WAY" type="int" value="0"/>

  </node>

  #### Asctec Proc ########################################

  <node pkg="nodelet" type="nodelet" name="AsctecProcNodelet" 
    args="load asctec_proc/AsctecProcNodelet pelican_manager" output="screen">


  </node>

  #### Waypoint Manager ###################################
  
  <node pkg="asctec_proc" type="waypoint_manager_absolute.py" name="WaypointManager" 
  args="load asctec_proc/AsctecProcNodelet pelican_manager" launch-prefix="xterm -e">
  </node>

  #### Plot GPS DATA ######################################

  <node pkg="asctec_proc" type="plot_gps_data.py" name="PlotGpsData" 
  args="load asctec_proc/AsctecProcNodelet pelican_manager">
  </node>

</launch>



